W-12483568 update chart values and other mods duke (#217)
mulesoft/docs-gateway: gateway/1.4/modules/ROOT/pages/flex-gateway-k8-change-helm-settings.adoc

= Update Pod Settings for a Flex Gateway Deployment Through a Helm Chart

Change requirements for a Pod that contains a Flex Gateway deployment. The example in this guide sets the following CPU and memory requirements in the Helm chart for a Flex Gateway deployment:

* `resources.limits.cpu=1500m`
* `resources.limits.memory=2048Mi`
* `resources.requests.cpu=260m`
* `resources.requests.memory=384Mi`

> Use this example as a guide to updating any values in the Helm chart for your Flex Gateway deployment. 

Reviewer: ```suggestion
Use this example as a guide to updating values in the Helm chart for your Flex Gateway deployment. 
```

Find the default Helm chart settings for Flex Gateway through one of the following methods:

> 

Reviewer: A more direct way to say this:

Open...
Run...

Author: Updated, though I do find that the gerund emphasizes the optional nature of the methods. For similar reasons, I've always disagreed with imperative title styles like "Do X" vs. "Doing X". I don't need to run the steps in "Doing X" unless I have a reason to or the doc tells me to do it _for some reason_. And if "Do X" is optional, why would we tell them to do it via a title. It's a bit different here because the intro to the bullets says "one of the following methods".

> * Opening the *flex-gateway* page in https://artifacthub.io/packages/helm/flex-gateway/flex-gateway[ArtifactHUB^]

Author: ```suggestion
* Open the *flex-gateway* page in https://artifacthub.io/packages/helm/flex-gateway/flex-gateway[ArtifactHUB^]
```

> * Running `helm show values &lt;repository-name&gt;/&lt;chart-name&gt;` from a terminal window, for example, running `helm show values flex-gateway/flex-gateway`

Author: ```suggestion
* Run `helm show values &lt;repository-name&gt;/&lt;chart-name&gt;` from a terminal window, for example, running `helm show values flex-gateway/flex-gateway`
```

. Change the `cpu` and `memory` values by using the `helm upgrade` command.
+
.Syntax:
----
helm upgrade <release-name> <repository name>/<chart name> \
-i --wait -n <namespace> --create-namespace \
--set-file registration.content=registration.yaml \
--set <key-to-update>=<new-value> 
--set <next-key-to-update>=<new-value> 
...
----
+
The `helm upgrade` command relies on the `registration.yaml` file but does not add your new settings to this file. 
+
.Example:
[source,helm]
----
helm upgrade ingress flex-gateway/flex-gateway \
-i --wait -n gateway --create-namespace \
--set-file registration.content=registration.yaml \
--set resources.limits.cpu=1500m \
--set resources.limits.memory=2048Mi \
--set resources.requests.cpu=260m \
--set resources.requests.memory=384Mi
----
+
When successful, the command returns output similar to this:
+
----
Release "ingress" has been upgraded. Happy Helming!
NAME: ingress
LAST DEPLOYED: Mon Mar 27 11:58:36 2023
NAMESPACE: gateway
STATUS: deployed
REVISION: 3
TEST SUITE: None
----  

> . Using the new `REVISION` value, check the `resources` settings, for example:

Reviewer: More direct: Use the new REVISION` value to check...

+
[source,helm]
----
helm get values --revision=3 ingress -n gateway
----
+
The output displays the new `resources` settings for the Pod:
+
----
USER-SUPPLIED VALUES:
registration:
  ...
resources:
  limits:
    cpu: 1500m
    memory: 2048Mi
  requests:
    cpu: 260m
    memory: 384Mi
----
+

> Note that the `resources` settings are not displayed through the `registration.yaml`.

Reviewer: Suggestion: remove "Note that"
We try not to use admonitions unless they are really needed.

[[use-helm-f-option]]
== Provide a File with Modified Settings for Your Helm Chart

> After following the steps in <<update-cpu-memory-example>>, use the `--f` option with `helm upgrade` to pass a file that contains new Helm chart settings for your Flex Gateway deployment. Alternatively, follow the procedure in <<use-helm-set-option>>.

Reviewer: Pass a file where?

Author: Pass the file via the option. They use the -f (for file) option to pass the configuration file name to the command so that the configuration in the file can be incorporated into the Pod's settings.  "Pass"  is very common terminology that will be understood in the context of commands and command options. Are you thinking that it might be difficult to translate, so we should use something else?

Author: I updated it in case it helps with translation, but I don't think it was necessary, tbh.

. From the command line, navigate to the directory that contains the YAML registration file for your gateway release, such as a `registration.yaml` for an `ingress` release.
+
This file is generated when you register your gateway with Anypoint Platform in xref:flex-conn-reg-run.adoc[Connected Mode] or xref:flex-local-reg-run.adoc[Local Mode]. The file provides `registration.content` data for the `helm upgrade` step in this procedure. 

. Generate a configuration file that contains your new settings.
+
For example, use `echo` to create a configuration file named `my-config-file` that contains the new `cpu` and `memory` values:
+
[source,echo]
----
echo '{resources.limits.cpu: 1500m, resources.limits.memory:2048Mi, resources.requests.cpu: 260m, resources.requests.memory: 384Mi}' \
> my-config-file
----

. Pass the new values from your configuration file to your Helm chart, for example:
+

> .Syntax:

Reviewer: I don't think the Syntax label is needed here - especially because there are no other labels used for step 3.

Author: Ok, good catch! I instead added `.Example:` below `.Syntax:`, which I neglected to do earlier.

----
helm -n <namespace> upgrade <release-name> \
<repository name>/<chart name> -f <your-settings-config-file> \
--set-file registration.content=<registration-yaml-file>
----

> +

Author: ```suggestion
+
.Example:
```

[source,helm]
----
helm -n gateway upgrade ingress \
flex-gateway/flex-gateway -f my-config-file \
--set-file registration.content=registration.yaml
----
+
The `helm upgrade` command relies on the `registration.yaml` file but does not add your new settings to this file. 
+
When successful, the command returns output similar to this:
+
----
Release "ingress" has been upgraded. Happy Helming!
NAME: ingress
LAST DEPLOYED: Mon Mar 27 16:38:07 2023
NAMESPACE: gateway
STATUS: deployed
REVISION: 4
TEST SUITE: None
----

. Using your Helm repository and chart names, verify the update to your chart, for example:
+
[source,helm]
----
helm get values --revision=4 ingress -n gateway
----
+
The output displays the new `resources` settings for the Pod:
+
----
USER-SUPPLIED VALUES:
registration:
  ...
resources:
  limits:
    cpu: 1500m
    memory: 2048Mi
  requests:
    cpu: 260m
    memory: 384Mi
----
+

> Note that the `resources` settings are not displayed through the `registration.yaml`.

Reviewer: Same comment about admonitions.

+
.Example:
[source,helm]
----
helm upgrade ingress flex-gateway/flex-gateway \
-i --wait -n gateway --create-namespace \
--set-file registration.content=registration.yaml \
--set resources.limits.cpu=1500m \
--set resources.limits.memory=2048Mi \
--set resources.requests.cpu=260m \
--set resources.requests.memory=384Mi
----
+
When successful, the command returns output similar to this:
+
----
Release "ingress" has been upgraded. Happy Helming!
NAME: ingress
LAST DEPLOYED: Mon Mar 27 11:58:36 2023
NAMESPACE: gateway
STATUS: deployed
REVISION: 3
TEST SUITE: None
----  

. Using the new `REVISION` value, check the `resources` settings, for example:
+
[source,helm]
----
helm get values --revision=3 ingress -n gateway
----
+
The output displays the new `resources` settings for the Pod:
+
----
USER-SUPPLIED VALUES:
registration:
  ...
resources:
  limits:
    cpu: 1500m
    memory: 2048Mi
  requests:
    cpu: 260m
    memory: 384Mi
----
+
Note that the `resources` settings are not displayed through the `registration.yaml`.

[[use-helm-f-option]]
== Provide a File with Modified Settings for Your Helm Chart

> After following the steps in <<update-cpu-memory-example>>, use the `--f` option with `helm upgrade` to pass a file that contains new Helm chart settings for your Flex Gateway deployment. Alternatively, follow the procedure in <<use-helm-set-option>>.

Author: ```suggestion
After following the steps in <<update-cpu-memory-example>>, use the `--f` option with `helm upgrade` to identify the file that contains new Helm chart settings for your Flex Gateway deployment. Alternatively, follow the procedure in <<use-helm-set-option>>.
```
